tmc-langs-r/src/main/java/fi/helsinki/cs/tmc/langs/r/RStudentFilePolicy.java

```diff
@@ -0,0 +1,26 @@
+package fi.helsinki.cs.tmc.langs.r;
+
+import fi.helsinki.cs.tmc.langs.io.ConfigurableStudentFilePolicy;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class RStudentFilePolicy extends ConfigurableStudentFilePolicy {
+
+    public RStudentFilePolicy(Path configFileParent) {
+        super(configFileParent);
+    }
+
+    /**
+     * Returns {@code True} for all files in the <tt>projectRoot/R/</tt> directory and other
+     * files required for building the project.
+     *
+     * <p>Will NOT return {@code True} for any test files. If test file modification are part
+     * of the exercise, those test files are whitelisted as <tt>ExtraStudentFiles</tt> and the
+     * decision to include them is made by {@link ConfigurableStudentFilePolicy}.
+     */
+    @Override
+    public boolean isStudentSourceFile(Path path, Path projectRootPath) {
+        return path.startsWith(Paths.get("R"));
```

Comment:
So this would classify all the R files, including tests, as files that the student is allowed to edit. Is there some sort of source folder/test folder in your templates?

Reply:
At today's meeting, you expressed concern that the function would consider all folders whose names start with 'R' (for example. 'Rlol') to contain student files. We added a new test to check this, and it does not seem to be the case. The function correctly considers only the files in the directory which has the exact name 'R' to be student files.